snuba/processor.py

```diff
@@ -2,28 +2,31 @@
 import re
 from abc import ABC, abstractmethod
 from datetime import datetime, timedelta
-from enum import Enum
 from hashlib import md5
 from typing import Any, NamedTuple, Optional, Sequence, Union
 
 import simplejson as json
 
 from snuba.util import force_bytes
+from snuba.writer import WriterTableRow
+
 
 HASH_RE = re.compile(r"^[0-9a-f]{32}$", re.IGNORECASE)
 MAX_UINT16 = 2 ** 16 - 1
 MAX_UINT32 = 2 ** 32 - 1
 NIL_UUID = "00000000-0000-0000-0000-000000000000"
 
 
-class ProcessorAction(Enum):
-    INSERT = 0
-    REPLACE = 1
+class InsertBatch(NamedTuple):
+    rows: Sequence[WriterTableRow]
+
+
+class ReplacementBatch(NamedTuple):
+    key: str
+    values: Sequence[Any]
```

Comment:
This `Any` type will likely eventually be replaced with [`ReplacementMessage`](https://github.com/getsentry/snuba/blob/3f49a88722a0bda1652880450230232342493f22/snuba/replacers/replacer_processor.py#L8-L16) (or something that supersedes it.)